src/EventLoop/Internal/AbstractDriver.php

```diff
@@ -698,18 +711,25 @@ private function invokeMicrotasks(): void
                 }
 
                 if ($this->interrupt) {
-                    $interrupt = $this->interrupt;
-                    $this->interrupt = null;
-
-                    \Fiber::suspend($interrupt);
+                    $this->invokeInterrupt();
                 }
             }
         }
     }
 
-    private function interrupt(callable $callback): void
+    private function invokeInterrupt(): void
     {
-        $this->interrupt = $callback;
+        \assert($this->interrupt !== null);
+
+        $interrupt = $this->interrupt;
+        $this->interrupt = null;
+
+        if (!$this->inFiber) {
```

Comment:
Why do we need to make this conditional?

Reply:
If we're not in a fiber (such as in the tests) we want to simply throw the exception.

Reply:
So our own tests are the only case where this will be true?

Reply:
Generally yes. We could require the loop be run in a fiber, I wouldn't be against that, but perhaps another PR?

Reply:
Yes, let's do it in another PR.